lib/sycamore/sycamore/utils/cache.py

```diff
@@ -33,20 +34,23 @@ def set(self, hash_key: str, hash_value):
 
 
 class S3Cache(Cache):
-    def __init__(self, s3_client, s3_path: str, freshness_in_seconds: int = -1):
-        self._s3_client = s3_client
+    def __init__(self, s3_path: str, freshness_in_seconds: int = -1):
         self._s3_path = s3_path
         self._freshness_in_seconds = freshness_in_seconds
+        self._s3_client = None
 
     def _get_s3_bucket_and_key(self, key):
         parts = self._s3_path.replace("s3://", "").strip("/").split("/", 1)
         return parts[0], "/".join([parts[1], key]) if len(parts) == 2 else key
 
     def get(self, key: str):
+        if not self._s3_client:
```

Comment:
@bsowell let me know your thoughts on this. I chose not to use the clientwrapper interface you've used in `OpenAI` because it seemed counter-intuitive to have to pass a CacheProvider instead of a cache. Happy to revisit

Reply:
I guess the big question is whether you ever need more parameters to pass to the client. In the OpenAI case, there are a bunch of different options that we need to expose, so we would need to pickle those, regardless. Here, I guess, the question is whether we would want to customize how we get credentials. Probably not a big deal since you can always use env vars if you need to.

Reply:
Yeah that's fair, I think we will need to at some point if we're using fancier client configs. Fine to defer until then